addons/purchase/models/purchase.py

```diff
@@ -58,6 +58,8 @@ def _compute_invoice(self):
             order.invoice_count = len(invoices)
 
     READONLY_STATES = {
+        'draft': [('readonly', False)],
+        'sent': [('readonly', False)],
```

Comment:
You can also remove all these below as you have added readonly=True at field level.

For a stable Version, we should handle this at the XML level, not the field level as it affects DB schema.

Thanks